packages/block-editor/src/components/inner-blocks/index.js

```diff
@@ -49,24 +49,34 @@ function UncontrolledInnerBlocks( props ) {
 	} = props;
 
 	const isSmallScreen = useViewportMatch( 'medium', '<' );
-
-	const { hasOverlay, block, enableClickThrough } = useSelect( ( select ) => {
-		const {
-			getBlock,
-			isBlockSelected,
-			hasSelectedInnerBlock,
-			isNavigationMode,
-		} = select( 'core/block-editor' );
-		const theBlock = getBlock( clientId );
-		return {
-			block: theBlock,
-			hasOverlay:
-				theBlock.name !== 'core/template' &&
-				! isBlockSelected( clientId ) &&
-				! hasSelectedInnerBlock( clientId, true ),
-			enableClickThrough: isNavigationMode() || isSmallScreen,
-		};
-	} );
+	const { hasOverlay, isNavMode, context } = useSelect(
+		( select ) => {
+			const {
+				getBlock,
+				isBlockSelected,
+				hasSelectedInnerBlock,
+				isNavigationMode,
+			} = select( 'core/block-editor' );
+			const block = getBlock( clientId );
+			const blockType = getBlockType( block.name );
+			let blockContext;
+
+			if ( blockType && blockType.providesContext ) {
+				blockContext = getBlockContext( block.attributes, blockType );
+			}
+
+			return {
+				hasOverlay:
+					block.name !== 'core/template' &&
+					! isBlockSelected( clientId ) &&
+					! hasSelectedInnerBlock( clientId, true ),
+				isNavMode: isNavigationMode(),
+				blockContext,
+			};
+		},
+		[ clientId ]
+	);
+	const enableClickThrough = isNavMode && isSmallScreen;
```

Comment:
Is the change from `||` to `&&` intentional here?

Reply:
No, thanks for catching that!

Reply:
```suggestion
	const enableClickThrough = isNavMode || isSmallScreen;
```